src/datastore/postgres-store.ts

```diff
@@ -5860,11 +5860,12 @@ export class PgDataStore
       tx_id: Buffer;
     }>(
       `
-      SELECT locked_amount, unlock_height, block_height, tx_id, locked_address
-      FROM stx_lock_events
-      WHERE canonical = true AND unlock_height <= $1 AND unlock_height > $2
+      SELECT locked_amount, unlock_height, sle.block_height, locked_address, txs.tx_id
+      FROM stx_lock_events sle INNER JOIN txs ON ($3 = txs.block_height)
```

Comment:
I'm having a hard time reasoning about the correctness of this query, and we don't have tests that demonstrate much of the Stacks vs Bitcoin block height differences that the `unlock_height <= $1 AND unlock_height > $2` statement is responsible for. 

It looks like there's at least one issue here where the `txs` rows are not checked for `canonical = true AND microblock_canonical = true`. Perhaps it makes sense to preserve the original query, and create a new one to grab the correct coinbase tx?